### ISSUE TITLE
[FIRRTL] (tail) false negative

### ISSUE THREAD

**Issue by drom** [llvm/circt#1136]
The following FIRRTL program

```python
circuit top_mod :
  module top_mod :
    output tmp5: UInt<16>
    output tmp55: SInt
    tmp55 <= mux(head(UInt<11>(57), 1), SInt("h66b9c5"), SInt("b-1"))
    tmp5 <= tail(tmp55, 8)
```

Compiled with `firtool` produces this Error:

```
a_top_mod.fir:6:13: error: amount must be less than or equal operand width
    tmp5 <= tail(tmp55, 8)
            ^
```

Compiled with `firrtl-1.5-SNAPSHOT` produces this Verilog:

```verilog
module top_mod(
  output [15:0] tmp5,
  output [23:0] tmp55
);
  assign tmp5 = tmp55[15:0];
  assign tmp55 = -24'sh1;
endmodule
```

**Comment by fabianschuiki** [comment]
This will be fixed once #1238 lands (fixes the order in which ports and operations are inferred, among other stuff).
